### ISSUE TITLE
cannot connect to i3 window manager socket path atom library is null

### ISSUE THREAD

**Issue by psamim** [denesb/xfce4-i3-workspaces-plugin#30]
Hi,

I am not sure if this error is caused by `xfce4-i3-workspaces-plugin` or not. I start i3 and `xfce4-panel`  using `startx` without any display manager.

When I close i3, the message `cannot connect to i3 window manager socket path atom library is null` is displayed continuously. When I switch to another TTY and kill the process `/usr/lib/xfce4/panel/wrapper-1.0 /usr/lib/xfce4/panel/plugins/libi3workspaces.so`, the message is stopped.

Thanks,
Samim

**Comment by denesb** [comment]
Yes, the plugin tries to reconnect to the window manager continuously after a disconnect. I'm not sure what to do about that. This makes sense when you are running an i3 session and the window manager have restarted or crashed. You can remove the plugin from the panel when you are done using i3 and then the spamming will stop.

**Comment by psamim** [comment]
Maybe it can stop trying to reconnect after some time.

Or react differently on i3 exit command.

Thanks,
Samim

**Comment by denesb** [comment]
I have no way of determining how i3 exited. I could modify it to check only once in a second.

**Comment by psamim** [comment]
Frequency of checking does not matter. As long as it is checking, I cannot exit i3 and return to shell. I think we can have maximum retries.

**Comment by psamim** [comment]
But it seems that the ultimate solution is to have an exit event for i3ipc.

**Comment by psamim** [comment]
Actually there is a shutdown event.

https://github.com/acrisci/i3ipc-glib/issues/11#issuecomment-216283924

**Comment by acrisci** [comment]
I'm working in the [exit event](https://github.com/i3/i3/pull/2335). It's done, I just need to do some cleanup.

**Comment by rharish101** [comment]
@denesb Sorry for bumping this old issue, but i3 has solved this here: [https://i3wm.org/docs/ipc.html#_shutdown_event](https://i3wm.org/docs/ipc.html#_shutdown_event). Are you still willing to work on this? If yes, will you fix this issue using this shutdown event?

**Comment by liar666** [comment]
WFIW, this makes my ".xsession-errors.old" quickly fill the 37GB left of my /home partition :{

**Comment by hypergig** [comment]
This happened to me too.. after I restarted i3 to make a config change
